packages/form-core/src/utils.ts

```diff
@@ -376,3 +380,30 @@ export function shallow<T>(objA: T, objB: T) {
   }
   return true
 }
+
+/**
+ * Determines the logic for determining the error value to set on the field meta within the form sync validation.
+ * @private
+ */
+export const determineErrorValue = ({
+  newFormValidatorError,
+  isPreviousErrorFromFormValidator,
+  previousErrorValue,
+}: {
+  newFormValidatorError: ValidationError
+  isPreviousErrorFromFormValidator: boolean
+  previousErrorValue: ValidationError
+}) => {
```

Comment:
Return type could be an explicit `{ newErrorValue: ValidationError, newSource: ValidationSource | undefined }`